client_src/js/lib/octofarm_client.js

```diff
@@ -0,0 +1,42 @@
+export default class OctoFarmClient {
+  static async get(item) {
+    const url = `/${item}`;
+    const get = await fetch(url, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    if (get.status === 400) {
+      throw new Error(
+        `Malformed request! Status: ${get.status} - ${get.statusText}`
+      );
+    } else if (get.status === 503) {
+      throw new Error("Error contacting server, is it alive?");
+    } else if (get.status === 204) {
+      return;
+    }
+    return await get.json();
+  }
+  static async post(item, data) {
+    const url = `/${item}`;
+    const post = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
+    });
+    if (post.status === 400) {
+      throw new Error(
+        `Malformed request! Status: ${post.status} - ${post.statusText}`
+      );
+    } else if (post.status === 503) {
```

Comment:
Created a new OctoFarm client that does the status checks and then passes back json if available. Simplifying the code required to utilising it and stopping all the repeated status checks elseware. Plan to migrate rest of code in another PR but left the old one in for now.